src/main/java/io/neonbee/data/DataVerticle.java

```diff
@@ -343,11 +343,20 @@ public static <U> Future<U> requestData(Vertx vertx, DataRequest request, DataCo
                         LOGGER.correlateWith(context).debug("Received event bus reply");
 
                         if (asyncReply.succeeded()) {
-                            context.setData(Optional
-                                    .ofNullable(
-                                            decodeContextFromString(asyncReply.result().headers().get(CONTEXT_HEADER)))
-                                    .map(DataContext::data).orElse(null));
-                            return succeededFuture(asyncReply.result().body());
+                            U body = asyncReply.result().body();
+                            if (body instanceof DataException) {
+                                if (LOGGER.isWarnEnabled()) {
+                                    LOGGER.correlateWith(context).warn("Failed to receive event bus reply from {}",
```

Comment:
I think we can get a little more specific here, because it no longer means that the "receive failed", but maybe something along those likes of "Received a event bus reply failure from {}"